Sources/PluginLibrary/ProtoLanguage.swift

```diff
@@ -0,0 +1,42 @@
+// Sources/PluginLibrary/ProtoLanguage.swift - Proto language utilities
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+// -----------------------------------------------------------------------------
+///
+/// Utility functions for dealing with proto language syntax
+///
+// -----------------------------------------------------------------------------
+
+import Swift
+import Foundation
+
+public func isValidProtobufIdentifier(_ s: String) -> Bool {
```

Comment:
I share your misgivings about the valid protobuf test.  I may drop that.

For the Swift identifier test, I went with something more ambitious here primarily to make sure we're very clear about what this support code does.  I expect that test to get used a lot so I don't want there to be any confusion about it.